diff --git a/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs b/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs
index 2cb0cacd..63d671e1 100644
--- a/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs
+++ b/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs
@@ -131,7 +131,7 @@ namespace Microsoft.VisualStudio.Services.Agent
                 }
 
                 // this should give us _diag folder under agent root directory as default value for diagLogDirctory
-                string diagLogPath = GetDiagDirectory(_hostType);
+                string diagLogPath = GetDirectory(WellKnownDirectory.Diag);
                 _traceManager = new TraceManager(new HostTraceListener(diagLogPath, hostType.ToString(), logPageSize, logRetentionDays), this.SecretMasker);
 
             }
@@ -184,6 +184,12 @@ namespace Microsoft.VisualStudio.Services.Agent
                     path = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
                     break;
 
+                case WellKnownDirectory.Diag:
+                    path = Path.Combine(
+                        GetDirectory(WellKnownDirectory.Root),
+                        Constants.Path.DiagDirectory);
+                    break;
+
                 case WellKnownDirectory.Externals:
                     path = Path.Combine(
                         GetDirectory(WellKnownDirectory.Root),
@@ -266,7 +272,10 @@ namespace Microsoft.VisualStudio.Services.Agent
                     throw new NotSupportedException($"Unexpected well known directory: '{directory}'");
             }
 
-            _trace.Info($"Well known directory '{directory}': '{path}'");
+            if (_trace != null)
+            {
+                _trace.Info($"Well known directory '{directory}': '{path}'");
+            }
             return path;
         }
 
